packaging/resource_suite_s3_nocache.py

```diff
@@ -2198,6 +2198,73 @@ def test_iput_and_icp_with_invalid_secret_key__issue_6154(self):
+            self.admin.assert_icommand("iadmin mkresc %s replication" % repl_resource_name, 'STDOUT_SINGLELINE', "Creating")
+            self.admin.assert_icommand("iadmin mkresc %s s3 %s:/%s/%s %s" %
+                                   (s3_resc1_resource_name, hostname, self.s3bucketname, s3_resc1_resource_name, self.s3_context), 'STDOUT_SINGLELINE', "Creating")
+            self.admin.assert_icommand("iadmin mkresc %s s3 %s:/%s/%s %s" %
+                                   (s3_resc2_resource_name, hostname, self.s3bucketname, s3_resc2_resource_name, self.s3_context), 'STDOUT_SINGLELINE', "Creating")
+            self.admin.assert_icommand("iadmin mkresc %s s3 %s:/%s/%s %s" %
+                                   (s3_resc3_resource_name, hostname, self.s3bucketname, s3_resc3_resource_name, self.s3_context), 'STDOUT_SINGLELINE', "Creating")
+            self.admin.assert_icommand("iadmin addchildtoresc %s %s" % (repl_resource_name, s3_resc1_resource_name))
+            self.admin.assert_icommand("iadmin addchildtoresc %s %s" % (repl_resource_name, s3_resc2_resource_name))
+            self.admin.assert_icommand("iadmin addchildtoresc %s %s" % (repl_resource_name, s3_resc3_resource_name))
+
+            file_sizes = [
+                    100,
+                    4 * 1024 * 1024 - 1,
+                    4 * 1024 * 1024,
+                    8 * 1024 * 1024,
+                    16 * 1024 * 1024,
+                    32 * 1024 * 1024 - 1,
+                    32 * 1024 * 1024,
+                    ];
+
+            for file_size in file_sizes:
+                # create the file
+                lib.make_arbitrary_file(filename, file_size)
+
+                # put the file to repl_resource
+                self.user1.assert_icommand("iput -R %s %s" % (repl_resource_name, filename))
+
+                # debug
+                self.user1.assert_icommand("ils -l %s" % filename, 'STDOUT_SINGLELINE', filename)
+
+                # get each replica
+                self.user1.assert_icommand("iget -f -n 0 %s %s_get0" % (filename, filename))
+                self.user1.assert_icommand("diff %s %s_get0" % (filename, filename), 'EMPTY')
+                self.user1.assert_icommand("iget -f -n 1 %s %s_get1" % (filename, filename))
+                self.user1.assert_icommand("diff %s %s_get1" % (filename, filename), 'EMPTY')
+                self.user1.assert_icommand("iget -f -n 2 %s %s_get2" % (filename, filename))
+                self.user1.assert_icommand("diff %s %s_get2" % (filename, filename), 'EMPTY')
+
+                # cleanup before next loop
+                s3plugin_lib.remove_if_exists(filename)
+                s3plugin_lib.remove_if_exists("%s_get0" % filename)
+                s3plugin_lib.remove_if_exists("%s_get1" % filename)
+                s3plugin_lib.remove_if_exists("%s_get2" % filename)
+                self.user1.assert_icommand("irm -f %s" % (filename))
+
+        finally:
+            # clean up resource tree
```

Comment:
Might want to add another `irm -f` down here because it might not be able to clean up the resources if the test fails before the `irm -f` above.

Reply:
done